### PR TITLE
Fix #20332 - address fields sort order fixes

### LINKED ISSUE #20332: Shipping address form City, State/Province, Zip/Postal Code, Country not in right Order in checkout
Shipping address form **City, State/Province, Zip/Postal Code, Country fields** not in right Order in checkout 

### Preconditions (*)

1. Magento 2.3.0
2. PHP 7.2

### Steps to reproduce (*)

Step 1: Open frontend 
Step 2: Add any product in cart 
Step 3: Go to checkout and and see shipping address form

### Expected result (*)
Order should be in : **Country**, **State/Province**, **City**, **Zip/Postal Code fields**

sol

### Actual result (*)

arpklxz

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is fix for #20332

### Fixed Issues (if relevant)

1. magento/magento2#20332: Shipping address form City, State/Province, Zip/Postal Code, Country not in right Order in checkout

### Manual testing scenarios (*)

1. Go to any form that contains customer address fields
2. Four fields order should be changed to: **Country, State/Province, City, Zip/Postal Code** 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Checkout/Block/Checkout/LayoutProcessor.php [hunk 1]
```diff
@@ -45,39 +50,35 @@ class LayoutProcessor implements \Magento\Checkout\Block\Checkout\LayoutProcesso
...
         $this->merger = $merger;
-        $this->options = $options ?: \Magento\Framework\App\ObjectManager::getInstance()
-            ->get(\Magento\Customer\Model\Options::class);
-        $this->checkoutDataHelper = $checkoutDataHelper ?: \Magento\Framework\App\ObjectManager::getInstance()
-            ->get(Data::class);
-        $this->shippingConfig = $shippingConfig ?: \Magento\Framework\App\ObjectManager::getInstance()
-            ->get(\Magento\Shipping\Model\Config::class);
-        $this->storeManager = $storeManager ?: \Magento\Framework\App\ObjectManager::getInstance()
-            ->get(StoreManagerInterface::class);
+        $this->options = $options ?: ObjectManager::getInstance()->get(Options::class);
```
Comment (lenaorobei) [inline app/code/Magento/Checkout/Block/Checkout/LayoutProcessor.php]: `ObjectManager` is redundant here.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please address BC comment and pull latest changes from `2.4-develop` branch.

**Comment by lenaorobei** [comment]
@Bartlomiejsz could you please merge latest changes and address review comment?
Appreciate that!

File: app/code/Magento/Checkout/Block/Checkout/LayoutProcessor.php (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/Checkout/Block/Checkout/LayoutProcessor.php]: @lenaorobei done

**Comment by Bartlomiejsz** [comment]
@lenaorobei done, ready for review

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-6990](https://jira.corp.magento.com/browse/ENGCOM-6990) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**